components/dropdown/src/floatingUI.mjs

```diff
@@ -10,22 +10,53 @@ export default class AuroFloatingUI {
     this.keyDownHandler = null;
   }
 
+  mirrorSize(fullscreen) {
+    // mirror the boxsize from bibSizer
+    const sizerStyle = window.getComputedStyle(this.element.bibSizer);
+    const bibContent = this.element.bib.shadowRoot.querySelector(".container");
+    if (fullscreen) {
+      bibContent.style.width = '';
+      bibContent.style.height = '';
+      bibContent.style.maxWidth = '';
+      bibContent.style.maxHeight = '';
+    } else {
+      if (sizerStyle.width !== '0px') {
+        bibContent.style.width = sizerStyle.width;
+      }
+      if (sizerStyle.height !== '0px') {
+        bibContent.style.height = sizerStyle.height;
+      }
+      bibContent.style.maxWidth = sizerStyle.maxWidth;
+      bibContent.style.maxHeight = sizerStyle.maxHeight;
+    }
+  }
+
+  getPositioningStrategy() {
+    if (this.element.bib.mobileFullscreenBreakpoint) {
+      const isMobile = window.matchMedia(`(max-width: ${this.element.bib.mobileFullscreenBreakpoint})`).matches;
+      return isMobile ? 'fullscreen' : 'floating';
+    }
+    return 'floating';
+  }
+
   position() {
+    const strategy = this.getPositioningStrategy();
+    if (strategy === 'fullscreen') {
+      this.handleMobileFullscreen(true);
+      this.mirrorSize(true);
+      return;
+    }
+
+    this.handleMobileFullscreen(false);
+    this.mirrorSize(false);
+
     // Define the middlware for the floater configuration
     const middleware = [
       offset(this.element.floaterConfig.offset || 0),
+      ...(this.element.floaterConfig.flip ? [flip()] : []), // Add flip middleware if flip is enabled
+      ...(this.element.floaterConfig.autoPlacement ? [autoPlacement()] : []), // Add autoPlacement middleware if autoPlacement is enabled
```

Comment:
I would have thought that the falsy statement of the ternary functions would have reset the array to be empty rather than just not adding the configuration. Did you validate that this for sure works correctly?

Reply:
yes, it's validated. The ternary operator doesn’t reset or clear arrays automatically. because each line is to lay out the values of the array with `...()`, not directly setting any value.